src/js/controller/controller.js

```diff
@@ -256,19 +258,27 @@ Object.assign(Controller.prototype, {
                 return;
             }
 
-            // Detect and store browser autoplay setting in the model.
-            const adConfig = _this._model.get('advertising');
-            canAutoplay(mediaPool, {
-                cancelable: checkAutoStartCancelable,
-                muted: _this.getMute(),
-                allowMuted: adConfig ? adConfig.autoplayadsmuted : false
-            })
-                .then(result => _model.set('canAutoplay', result))
-                .catch(function() {});
+            if (_model.get('autostart')) {
+                // Detect and store browser autoplay setting in the model.
+                const adConfig = _this._model.get('advertising');
+                canAutoplay(mediaPool, {
+                    cancelable: checkAutoStartCancelable,
+                    muted: _this.getMute(),
+                    allowMuted: adConfig ? adConfig.autoplayadsmuted : true
+                }).then(result => {
+                    _model.set('canAutoplay', result);
 
-            if (!OS.mobile && _model.get('autostart') === true) {
-                // Autostart immediately if we're not mobile and not waiting for the player to become viewable first
-                _autoStart();
+                    // Autostart immediately if we're not waiting for the player to become viewable first.
+                    if (!_model.get('playOnViewable')) {
+                        _autoStart();
+                    }
+                }).catch(function() {
```

Comment:
Can we get the `error` and pass include it in the `AUTOSTART_NOT_ALLOWED` event? Is the only way that this throws is cancelling and timing out?

That means we get `AUTOSTART_NOT_ALLOWED` events under those conditions, and not when the autoplay test fails legitimately?

Reply:
Yes, only on cancel and timeout. Do you want the entire Error, or just error.message?

Reply:
https://github.com/jwplayer/jwplayer/pull/2743/files I've made all scenarios that trigger `AUTOSTART_NOT_ALLOWED` throw, and added 'reason' to the errors.